src/languages/perl.js

```diff
@@ -258,19 +261,44 @@ export default function(hljs) {
     end: /\}/
     // contains defined later
   };
-  const VAR = { variants: [
-    { begin: /\$\d/ },
-    { begin: regex.concat(
-      /[$%@](\^\w\b|#\w+(::\w+)*|\{\w+\}|\w+(::\w*)*)/,
-      // negative look-ahead tries to avoid matching patterns that are not
-      // Perl at all like $ident$, @ident@, etc.
-      `(?![A-Za-z])(?![@$%])`
-    ) },
-    {
-      begin: /[$%@][^\s\w{]/,
-      relevance: 0
-    }
-  ] };
+  const ATTRS = {
+    className: 'attribute',
+    begin: /\s+:\s*\w+/,
+    end: /(\\s*\\(.*?\\))?/,
+  };
+  const VAR = {
+    variants: [
+      { begin: /\$\d/ },
+      { begin: regex.concat(
+        /[$%@](\^\w\b|#\w+(::\w+)*|\{\w+\}|\w+(::\w*)*)/,
+        // negative look-ahead tries to avoid matching patterns that are not
+        // Perl at all like $ident$, @ident@, etc.
+        `(?![A-Za-z])(?![@$%])`
+        )
+      },
+      {
+        begin: /[$%@][^\s\w{]/,
+        relevance: 0
+      }
+    ],
+    contains: [ ATTRS ],
+  };
+  const NUMBER = {
+    className: 'number',
+    variants: [
+      // in Perl numbers can have '_' in the middle, but not at the end,
+      // so we lookbehind the '\b' for a non-trailing '_'.
+      // decimal numbers
+      { match: /0?\.[0-9][0-9_]+(?<!_)\b/ },
```

Comment:
We can't use look-behind yet, it's too newly supported in Safari. Sorry.

Reply:
Would these work mostly OK if we just removed this check?

Reply:
Hmm, that's not considered an error at runtime, just a warning, so removing the lookbehind should be fine, or as you said "mostly ok" for now :) I could change the regex to `0?\.[0-9][0-9_]+[^_\W]\b` to something similar, but lets just remove it, as it's just a warning in the linter.

Reply:
Ah, but if you're suggesting to completely remove the `match`, it gets the specific case of a number starting with a dot, eg. `.32` is translated to `0.32` into the language. Keeping the match separated guarantees the period will be highlighted as a single number, like `4.23` would.

I could certainly include it in the second match, but it would get quite confusing to understand. I'll improve the comments there for future ref.